models/classes/theme/SolarDesignCheckerTrait.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\theme;
+
+trait SolarDesignCheckerTrait
+{
+    public function isSolarDesignEnabled(): bool
+    {
+        $themeData = $this->getThemeData();
+        if (array_key_exists(SolarDesignCheckerInterface::SOLAR_DESIGN_ENABLED, $themeData) &&
+            $themeData[SolarDesignCheckerInterface::SOLAR_DESIGN_ENABLED]) {
+            return true;
+        }
+        return false;
```

Comment:
chore: Can be simplified:

```suggestion
        $themeData = $this->getThemeData();

        return !empty($themeData[SolarDesignCheckerInterface::SOLAR_DESIGN_ENABLED]);
```

It should not generate an error if the index does not exits. 

**Personal opinion**: I would not create this trait (or interface). I think adding a new method `isSolarDesignEnabled` in the `ConfigurableTheme` should be enough.

That being said, we could them remove this from here https://github.com/oat-sa/tao-core/pull/4006/files#diff-a92025ec366e77d37bd57274d97219fcdb7de5aec058033d10c90f7a9f277c73R124

and let the only the `ConfigurableTheme` carry the business logic to decide if `SOLAR_DESIGN_ENABLED` or not. 

I am approving the PR as this is my personal view to have less code and a single point of maintenance.

Reply:
Thank you. Indeed the condition can be simplified and the `empty()` approach is even better as it relies on the implicit handling of values (less code = less bug)

Regarding the trait, the purpose is to remove the duplicates as the code is used in multiple places.